app/app_client.go

```diff
@@ -0,0 +1,1126 @@
+	})
+	if err != nil {
+		return "", err
+	}
+	return resp.UserId, nil
+}
+
+// CreateOrganization creates a new organization.
+func (c *AppClient) CreateOrganization(ctx context.Context, name string) (*Organization, error) {
+	resp, err := c.client.CreateOrganization(ctx, &pb.CreateOrganizationRequest{
+		Name: name,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return organizationFromProto(resp.Organization), nil
+}
+
+// ListOrganizations lists all the organizations.
+func (c *AppClient) ListOrganizations(ctx context.Context) ([]*Organization, error) {
+	resp, err := c.client.ListOrganizations(ctx, &pb.ListOrganizationsRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	var organizations []*Organization
+	for _, org := range resp.Organizations {
+		organizations = append(organizations, organizationFromProto(org))
+	}
+	return organizations, nil
+}
+
+// GetOrganizationsWithAccessToLocation gets all the organizations that have access to a location.
+func (c *AppClient) GetOrganizationsWithAccessToLocation(ctx context.Context, locationID string) ([]*OrganizationIdentity, error) {
+	resp, err := c.client.GetOrganizationsWithAccessToLocation(ctx, &pb.GetOrganizationsWithAccessToLocationRequest{
+		LocationId: locationID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var organizations []*OrganizationIdentity
+	for _, org := range resp.OrganizationIdentities {
+		organizations = append(organizations, organizationIdentityFromProto(org))
+	}
+	return organizations, nil
+}
+
+// ListOrganizationsByUser lists all the organizations that a user belongs to.
+func (c *AppClient) ListOrganizationsByUser(ctx context.Context, userID string) ([]*OrgDetails, error) {
+	resp, err := c.client.ListOrganizationsByUser(ctx, &pb.ListOrganizationsByUserRequest{
+		UserId: userID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var organizations []*OrgDetails
+	for _, org := range resp.Orgs {
+		organizations = append(organizations, orgDetailsFromProto(org))
+	}
+	return organizations, nil
+}
+
+// GetOrganization gets an organization.
+func (c *AppClient) GetOrganization(ctx context.Context, orgID string) (*Organization, error) {
+	resp, err := c.client.GetOrganization(ctx, &pb.GetOrganizationRequest{
+		OrganizationId: orgID,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return organizationFromProto(resp.Organization), nil
+}
+
+// GetOrganizationNamespaceAvailability checks for namespace availability throughout all organizations.
+func (c *AppClient) GetOrganizationNamespaceAvailability(ctx context.Context, namespace string) (bool, error) {
+	resp, err := c.client.GetOrganizationNamespaceAvailability(ctx, &pb.GetOrganizationNamespaceAvailabilityRequest{
+		PublicNamespace: namespace,
+	})
+	if err != nil {
+		return false, err
+	}
+	return resp.Available, nil
+}
+
+// UpdateOrganization updates an organization.
+func (c *AppClient) UpdateOrganization(ctx context.Context, orgID string, name, namespace, region, cid *string) (*Organization, error) {
```

Comment:
So, this `*string`-represents-an-optionally-passed-string pattern definitely _works_, but it's not the idiomatic one in Go. The idiomatic pattern is called "functional options," and it's kind of a pain to implement, but it's generally much easier for users. I'm gonna pull down your branch and try to give an example of what they look like for `UpdateOrganization`. If this change is super urgent, then no need to wait for me, but I think Go devs are gonna be surprised by `*string` params.